packages/api/src/payments/paymentProvider.ts

```diff
@@ -94,6 +101,53 @@ export abstract class BasePaymentProvider implements PaymentProvider {
   abstract createIntent(props: CreatePaymentIntentProps): Promise<Intent>
 
   abstract checkIntentStatus(props: CheckIntentProps): Promise<IntentState>
+
+  async updatePaymentWithIntentState({
+    intentState,
+    context
+  }: UpdatePaymentWithIntentStateProps): Promise<Payment> {
+    const payment = await context.loaders.paymentsByID.load(intentState.paymentID)
+    // TODO: should we overwrite already paid/canceled payments
+    if (!payment) throw new Error(`Payment with ID ${intentState.paymentID} not found`)
+
+    const updatedPayment = await context.dbAdapter.payment.updatePayment({
+      id: payment.id,
+      input: {
+        state: intentState.state,
+        paymentData: intentState.paymentData,
+        intentData: payment.intentData,
+        intentSecret: payment.intentSecret,
+        intentID: payment.intentID,
+        invoiceID: payment.invoiceID,
+        paymentMethodID: payment.paymentMethodID
+      }
+    })
+
+    if (!updatedPayment) throw new Error('Error while updating Payment')
+
+    if (intentState.customerID && payment.invoiceID) {
+      const invoice = await context.loaders.invoicesByID.load(payment.invoiceID)
+      if (!invoice?.userID)
+        throw new Error(`Invoice with ID ${payment.invoiceID} does not have a userID`)
+
+      const user = await context.dbAdapter.user.getUserByID(invoice.userID)
```

Comment:
Shouldn't `get` always return value otherwise throw? Wdyt about convention `get` returns object/throws, `find` returns object/null.

Reply:
I like that. Same comment as above. Something for the database abstraction layer discussion.